discovery/src/main/scala/akka/discovery/ServiceDiscovery.scala

```diff
@@ -18,30 +18,21 @@ final class ServiceDiscovery(implicit system: ExtendedActorSystem) extends Exten
       case method ⇒ method
     }
 
-  private lazy val _simpleImpl = ServiceDiscovery.loadServiceDiscovery(_simpleImplMethod, system)
+  private lazy val _simpleImpl = loadServiceDiscovery(_simpleImplMethod)
 
   /**
    * Default [[SimpleServiceDiscovery]] as configured in `akka.discovery.method`.
    */
   @throws[IllegalArgumentException]
   def discovery: SimpleServiceDiscovery = _simpleImpl
 
-}
-
-object ServiceDiscovery extends ExtensionId[ServiceDiscovery] with ExtensionIdProvider {
-  override def apply(system: ActorSystem): ServiceDiscovery = super.apply(system)
-
-  override def lookup: ServiceDiscovery.type = ServiceDiscovery
-
-  override def get(system: ActorSystem): ServiceDiscovery = super.get(system)
-
-  override def createExtension(system: ExtendedActorSystem): ServiceDiscovery = new ServiceDiscovery()(system)
-
   /**
-   * INTERNAL API
+   * Create a [[SimpleServiceDiscovery]] from configuration property.
+   * The given `method` parameter is used to find configuration property
+   * "akka.discovery.[method].class" or "[method].class". `method` can also
+   * be a fully class name.
    */
-  @InternalApi
-  private[akka] def loadServiceDiscovery(method: String, system: ExtendedActorSystem): SimpleServiceDiscovery = {
+  def loadServiceDiscovery(method: String): SimpleServiceDiscovery = {
```

Comment:
yes, I was thinking of that when we discussed how this could be used in the Play dependency injection and config. I agree that it should probably return same instance for the same `method`. Let me add that before merging.

Reply:
I would prefer to make that the job of whoever calls this. In many cases it would just be called once and reused anyway.

Reply:
If we have 2 different gRPC clients in Play and you configure them both with the same "method", don't we want those to use the same instance? Ok, if the `SimpleServiceDiscovery` is defined by it's own DI point it will only be one, but that wasn't how @raboof described that the current Play client DI is generated. Perhaps it should?